app/utils/gameUtils.ts

```diff
@@ -38,14 +38,44 @@ const getSpecialCardRank = (card: CardProps): number => {
 };
 
 /**
- * Checks if the given cards constitute a "flor" (all cards have the same suit)
+ * Checks if the given cards constitute a "flor" according to Uruguayan truco rules
  * @param cards The cards to check
- * @returns True if all cards have the same suit
+ * @param muestraCard The muestra card for special flor rules (optional)
+ * @returns True if the cards constitute a flor
  */
-export const hasFlor = (cards: CardProps[]): boolean => {
+export const hasFlor = (cards: CardProps[], muestraCard?: CardProps): boolean => {
```

Comment:
Please make the `muestraCard` param required. Let's remove the logic for cases where this param is not present (not longer a realistic case)